python/timeplus/source.py

```diff
@@ -0,0 +1,216 @@
+        """
+        body = {
+            # "name": self._name,
+            # "type": self._type,
+            # "stream": self._stream,
+            # "properties": self._properties
+        }
+
+        if self._name:
+            body["name"] = self._name
+
+        if self._type:
+            body["type"] = self._type
+
+        if self._stream:
+            body["stream"] = self._stream
+
+        if self._properties:
+            body["properties"] = self._properties
+
+        if self._description:
+            body["description"] = self._description
+
+        try:
+            self._metadata = self._api_instance.v1beta2_sources_post(body)
+            return self
+        except ApiException as e:
+            print("Exception when calling SourcesV1beta2Api->v1beta2_sources_post: %s\n" % e)
+            raise e
+
+    def list(self):
+        """
+        Fetches a list of all sources from the API.
+
+        Returns:
+        List: A list of all sources.
+
+        Raises:
+        ApiException: If an error occurs during the API call.
+        """
+        try:
+            list_response = self._api_instance.v1beta2_sources_get()
+            return list_response
+        except ApiException as e:
+            print("Exception when calling SourcesV1beta2Api->v1beta2_sources_get: %s\n" % e)
+            raise e
+
+    def delete(self, source_id):
+        """
+        Sends a request to the API to delete the source.
+
+        Args:
+        source_id (str): The id of the source to delete.
+
+        Raises:
+        ApiException: If an error occurs during the API call.
+        """
+        self._api_instance.v1beta2_sources_id_delete(source_id)
+
+    def get(self, source_id):
+        """
+        Retrieves the metadata of the source from the API.
+
+        Args:
+        source_id (str): The id of the source to retrieve.
+
+        Returns:
+        Source: The current source object with its metadata.
+
+        Raises:
+        TimeplusAPIError: If the source does not exist.
+        """
+        try:
+            self._metadata = self._api_instance.v1beta2_sources_id_get(source_id)
+            return self
+        except ApiException as e:
+            print("Exception when calling SourcesV1beta2Api->v1beta2_sources_id_get: %s\n" % e)
+            raise TimeplusAPIError(f"no such source with id {source_id}")
+
+    def metadata(self):
+        """
+        Returns the metadata of the source.
+
+        Returns:
+        Any: The metadata of the source.
+        """
+        return self._metadata
+
+    def update(self, source_id, body):
+        """
+        Update the specific source with the given ID.
+
+        Args:
+        source_id (str): The id of the source to update.
+        body (dict): A dictionary containing the parameters to update.
+
+        Raises:
+        ApiException: If an error occurs during the API call.
+        """
+        self._api_instance.v1beta2_sources_id_put(source_id, body)
+
```

Comment:
we need an id() function to get/set the id

you can refer to this set/get pattern
```
class MyClass:
    def __init__(self):
        self._value = None
    
    def value(self, *args):
        if len(args) == 0:
            print("Getting value...")
            return self._value
        elif len(args) == 1:
            new_value = args[0]
            print("Setting value...")
            self._value = new_value
        else:
            raise ValueError("value() accepts at most 1 argument")
```